### ISSUE TITLE
Only XML files are generated

### ISSUE THREAD

**Issue by vmag** [cloudflare/flan#9]
Only XML file is generated, IP for tex generator is empty:
```
docker run --name flan_1574405167 -v /srv/git/flan/shared:/shared flan_scan
# Nmap 7.70 scan initiated Fri Nov 22 06:46:08 2019 as: nmap -sV -oX /shared/xml_files/2019.11.22-06.46/x.x.x.18.xml -oN - -v1 --script=vulners/vulners.nse x.x.x.18
Nmap scan report for static.x-x-x-18.clients.your-server.de (x.x.x.18)
Host is up (0.00027s latency).
Not shown: 996 closed ports
PORT     STATE    SERVICE    VERSION
22/tcp   open     ssh        OpenSSH 7.4 (protocol 2.0)
| vulners: 
|   cpe:/a:openbsd:openssh:7.4: 
|       CVE-2018-15919  5.0     https://vulners.com/cve/CVE-2018-15919
|_      CVE-2017-15906  5.0     https://vulners.com/cve/CVE-2017-15906
80/tcp   open     http       nginx 1.16.1
|_http-server-header: nginx/1.16.1
443/tcp  open     ssl/http   nginx 1.16.1
|_http-server-header: nginx/1.16.1
|_http-trane-info: Problem with XML parsing of /evox/about
9001/tcp filtered tor-orport

Read data files from: /usr/bin/../share/nmap
Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
# Nmap done at Fri Nov 22 06:46:23 2019 -- 1 IP address (1 host up) scanned in 15.41 seconds
# Nmap 7.70 scan initiated Fri Nov 22 06:46:23 2019 as: nmap -sV -oX /shared/xml_files/2019.11.22-06.46/.xml -oN - -v1 --script=vulners/vulners.nse
Read data files from: /usr/bin/../share/nmap
WARNING: No targets were specified, so 0 hosts scanned.
# Nmap done at Fri Nov 22 06:46:23 2019 -- 0 IP addresses (0 hosts up) scanned in 0.35 seconds
WARNING: No targets were specified, so 0 hosts scanned.
Traceback (most recent call last):
  File "/output_report.py", line 225, in <module>
    main()
  File "/output_report.py", line 216, in main
    parse_results(data)
  File "/output_report.py", line 98, in parse_results
    hosts = data['nmaprun']['host']
KeyError: 'host'
sed: /shared/reports/report_2019.11.22-06.46.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-06.46.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-06.46.tex: No such file or directory
```

**Comment by Core269** [comment]
Same issue here;
`Traceback (most recent call last):
  File "/output_report.py", line 225, in <module>
    main()
  File "/output_report.py", line 216, in main
    parse_results(data)
  File "/output_report.py", line 98, in parse_results
    hosts = data['nmaprun']['host']
KeyError: 'host'
sed: /shared/reports/report_2019.11.22-10.39.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-10.39.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-10.39.tex: No such file or directory`

**Comment by opsi-math** [comment]
Similar issue, .xml report is unhelpful, Latex report not generated.


Read data files from: /usr/bin/../share/nmap
Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
# Nmap done at Fri Nov 22 20:10:38 2019 -- 256 IP addresses (21 hosts up) scanned in 3437.68 seconds
# Nmap 7.70 scan initiated Fri Nov 22 20:10:38 2019 as: nmap -sV -oX /shared/xml_files/2019.11.22-19.13/.xml -oN - -v1 --script=vulners/vulners.nse
Read data files from: /usr/bin/../share/nmap
WARNING: No targets were specified, so 0 hosts scanned.WARNING: No targets were specified, so 0 hosts scanned.

# Nmap done at Fri Nov 22 20:10:39 2019 -- 0 IP addresses (0 hosts up) scanned in 0.73 seconds
Traceback (most recent call last):
  File "/output_report.py", line 227, in <module>
    main()
  File "/output_report.py", line 218, in main
    parse_results(data)
  File "/output_report.py", line 98, in parse_results
    hosts = data['nmaprun']['host']
KeyError: 'host'
sed: /shared/reports/report_2019.11.22-19.13.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-19.13.tex: No such file or directory
sed: /shared/reports/report_2019.11.22-19.13.tex: No such file or directory

**Comment by TravelinMax** [comment]
I am having the same issue, but are you sure the xml file isn’t useful? Search the file for “cvss” or “cve” and you’ll see all your results.

**Comment by nadinelyab** [comment]
Hello, I believe this was happening because you were running flan scan with no IP addresses or with an empty line in your `ips.txt`. I have added a check to avoid this error. However if you run flan scan with a `ips.txt` file with just one empty line and no IP addresses you will get a `.tex` file that does not compile. I think this is okay for now since this is not the intended way of using Flan Scan. Let me know if you have more questions.

**Comment by wheelq** [comment]
@nadinelyab you believe or have you verified this?

**Comment by nadinelyab** [comment]
@wheelq I tested with an empty line in the `ips.txt` file and got the same error as above. Are you still having issues?

**Comment by wheelq** [comment]
Yes I am. I have added 1,2,5..20 lines of hosts untill it started to fail.
Then I have removed some hosts from the top, and the scanner did not fail anymore, meaning the issue is not with the last line. Then I have removed some more hosts from the top and the scanner started to fail again. So I think the issue is somewhere else. Any idea what to debug first instead of wild guessing and wasting time on assumptions? In my case tex files have got their own filenames, but for some reason scanner is having some issue with those:
```

KeyError: 'host'
sed: /shared/reports/report_2019.11.26-19.47.tex: No such file or directory
sed: /shared/reports/report_2019.11.26-19.47.tex: No such file or directory
sed: /shared/reports/report_2019.11.26-19.47.tex: No such file or directory
sed: /shared/reports/report_2019.11.26-19.47.tex: No such file or directory
```
